core/shared/src/main/scala/zio/ZPool.scala

```diff
@@ -175,7 +175,9 @@ object ZPool {
                       case Exit.Success(item) =>
                         invalidated.get.flatMap { set =>
                           if (set.contains(item))
-                            state.update(state => state.copy(free = state.free + 1)) *>
```

Comment:
Possible to call release on the Attempted to eliminate duplication / unify code paths?

Reply:
Yes!